pkg/pillar/cmd/upgradeconverter/convertglobalconfig.go

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2017-2018 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package upgradeconverter
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/lf-edge/eve/pkg/pillar/types"
+	log "github.com/sirupsen/logrus"
+	"io/ioutil"
+	"os"
+)
+
+func handleUpgradeGlobalConfig(action ConverterAction, oldFile string, newFile string) (err error) {
+	switch action {
+	case Convert:
+		err = convert(oldFile, newFile)
+	case CleanupNew:
+		err = cleanupNew(oldFile, newFile)
+	case CleanupOld:
+		err = cleanupOld(oldFile)
+	case NoAction:
+		break
+	}
+	return err
+}
+
+func convert(oldVersionPath string, newVersionPath string) (err error) {
+	oldVersionExists := fileExists(oldVersionPath)
+	newVersionExists := fileExists(newVersionPath)
+	if newVersionExists {
+		log.Infof("Confirmed that new version exists")
+		if oldVersionExists {
+			log.Infof("Confirmed that old version exists")
+			oldVersionTime, oldTimeErr := fileTimeStamp(oldVersionPath)
+			newVersionTime, newTimeErr := fileTimeStamp(newVersionPath)
+			if oldTimeErr == nil {
+				if newTimeErr == nil {
+					if oldVersionTime.After(newVersionTime) {
+						log.Infof("Deleting new version")
+						err = deleteFile(newVersionPath)
+					} else {
+						err = deleteFile(oldVersionPath)
+						return err
+					}
+				} else {
+					log.Errorf("failed to get a time stamp for new version file err %s", newTimeErr)
+				}
+			} else {
+				log.Errorf("failed to get a time stamp for old version file err %s", oldTimeErr)
+			}
+
+		} else {
+			return nil
+		}
+	}
+
+	if oldVersionExists {
+		oldVersion, openErr := os.Open(oldVersionPath)
+		if openErr != nil {
+			log.Errorf("failed to open old version err %s", openErr)
+		}
+		byteValue, readErr := ioutil.ReadAll(oldVersion)
+		if readErr != nil {
+			log.Errorf("failed to read old version err %s", readErr)
+		}
+		var oldGlobalConfig types.OldGlobalConfig
+		err = json.Unmarshal([]byte(byteValue), &oldGlobalConfig)
+		if err != nil {
+			err = fmt.Errorf("failed to unmarshall old config defaults were applied err %s", err)
+			oldGlobalConfig = types.ApplyDefaults(oldGlobalConfig)
```

Comment:
No, instead act as if the file didn't exist.
The new config will have the defaults.